core/src/main/java/org/elasticsearch/rest/action/admin/indices/RestGetAllMappingsAction.java

```diff
@@ -0,0 +1,107 @@
+import org.elasticsearch.common.collect.ImmutableOpenMap;
+import org.elasticsearch.common.settings.IndexScopedSettings;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.settings.SettingsFilter;
+import org.elasticsearch.common.xcontent.ToXContent.Params;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.rest.BaseRestHandler;
+import org.elasticsearch.rest.BytesRestResponse;
+import org.elasticsearch.rest.RestController;
+import org.elasticsearch.rest.RestRequest;
+import org.elasticsearch.rest.RestResponse;
+import org.elasticsearch.rest.action.RestBuilderListener;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Set;
+
+import static org.elasticsearch.rest.RestRequest.Method.GET;
+import static org.elasticsearch.rest.RestRequest.Method.HEAD;
+import static org.elasticsearch.rest.RestStatus.OK;
+
+/**
+ * The REST handler for retrieving all mappings
+ */
+public class RestGetAllMappingsAction extends BaseRestHandler {
+
+    public RestGetAllMappingsAction(final Settings settings, final RestController controller) {
+        super(settings);
+        controller.registerHandler(GET, "/_mapping", this);
+        controller.registerHandler(GET, "/_mappings", this);
+    }
+
+    @Override
+    public RestChannelConsumer prepareRequest(final RestRequest request, final NodeClient client) throws IOException {
+        final GetIndexRequest getIndexRequest = new GetIndexRequest();
+        getIndexRequest.indices(Strings.EMPTY_ARRAY);
+        getIndexRequest.features(Feature.MAPPINGS);
+        getIndexRequest.indicesOptions(IndicesOptions.fromRequest(request, getIndexRequest.indicesOptions()));
+        getIndexRequest.local(request.paramAsBoolean("local", getIndexRequest.local()));
+        getIndexRequest.humanReadable(request.paramAsBoolean("human", false));
+        return channel -> client.admin().indices().getIndex(getIndexRequest, new RestBuilderListener<GetIndexResponse>(channel) {
+
+            @Override
+            public RestResponse buildResponse(final GetIndexResponse response, final XContentBuilder builder) throws Exception {
+                builder.startObject();
+                {
+                    for (final String index : response.indices()) {
+                        builder.startObject(index);
+                        {
+                            writeMappings(response.mappings().get(index), builder);
+                        }
+                        builder.endObject();
+
+                    }
+                }
+                builder.endObject();
+
+                return new BytesRestResponse(OK, builder);
+            }
+
+            private void writeMappings(final ImmutableOpenMap<String, MappingMetaData> mappings,
+                                       final XContentBuilder builder) throws IOException {
+                builder.startObject("mappings");
+                {
+                    if (mappings != null) {
```

Comment:
Under what circumstances would the mappings for an index be null (as opposed to an empty map)?  It seems the default for `GetIndexResponse` is to always have an empty map for mappings (and aliases and settings) and it would only get assigned to a non-null map.